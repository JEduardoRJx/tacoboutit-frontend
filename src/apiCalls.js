export const getRestaurants = async (lat = 39.7392, lng = -104.9903) => {
  const response = await fetch(`https://tacoboutit-backend.herokuapp.com/api/v1/restaurants/retrieve/?lat=${lat}&lng=${lng}`);
  if(!response.ok) {
    throw Error('Failed to fetch restaurants near you');
  }
  return response.json();
}

export const newTaco = async (type, restaurant) => {
  const response = await fetch('https://tacoboutit-backend.herokuapp.com/api/v1/tacos/new/', {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify({ type, restaurant }),
  });
  if(!response.ok) {
    return Error('Failed to add taco.');
  }
  const resp = await response.json();
  return resp;
}

export const addReview = async (tacoId, rating, review = 'No Review Available') => {
  const response = await fetch(`https://tacoboutit-backend.herokuapp.com/api/v1/reviews/`, {
    method: 'POST', 
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify({
      taco: tacoId,
      rating,
      review,
    }),
  })
  if(!response.ok) {
    return Error('Failed to add review');
  }
  const resp = await response.json();
  return resp;
}
