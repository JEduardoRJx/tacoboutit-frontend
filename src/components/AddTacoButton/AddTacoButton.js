import React, { Component } from 'react'
import { Text, View, StyleSheet, TouchableOpacity, Modal, TextInput, Button, Picker, Dimensions } from 'react-native'

export default class AddTacoButton extends Component {
  state = { 
    showAddTaco: false,
    selectedType: 'carne asada',
  };

  handleSubmit = async () => {
    const { submitTaco, id } = this.props;
    const selectedType = this.state.selectedType;
    const result = await submitTaco(selectedType, id);
    this.setState({ showAddTaco: false });
  }

  render() {
    return (
      <TouchableOpacity style={styles.addTacoButton} onPress={() => this.setState({showAddTaco: !this.state.showAddTaco })}>
        <Text style={styles.addTacoText}>+ Taco</Text>
        <Modal 
          transparent={true}
          visible={this.state.showAddTaco} 
          animationType="slide"
          onRequestClose={() => this.setState({ showAddTaco: false })}
        >
          <View style={styles.addTacoModal} >
            <Text style={{fontSize: 24}}>Add New Taco</Text>
            <Text style={{fontSize: 24}}>Select Type</Text>
            <Picker
              selectedValue={this.state.selectedType}
              style={{height: 50, width: '100%'}}
              onValueChange={(itemValue, itemIndex) => {
                this.setState({selectedType: itemValue});
                console.log(this.state.selectedType);
              }
              }>
              <Picker.Item label="Carne Asada" value="carne asada" />
              <Picker.Item label="Al Pastor" value="al pastor" />
              <Picker.Item label="Cabeza" value="cabeza" />
              <Picker.Item label="Cabeza" value="cabeza" />
              <Picker.Item label="Carnitas" value="carnitas" />
            </Picker>
            <TouchableOpacity 
              onPress={() => this.handleSubmit()}>
                <View style={styles.buttonStyle}>
                  <Text style={{color: 'white', fontSize: 16}}>Submit</Text>
                </View>
              </TouchableOpacity>
          </View>
        </Modal>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  addTacoButton: {
    width: 100, 
    height: 40, 
    backgroundColor: 'lightgray',
    borderRadius: 20,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    zIndex: 5,
    top: 220,
    right: 10,

  },
  addTacoText: {
    fontSize: 20,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  addTacoModal: {
    width: Dimensions.get('window').width * 0.8, 
    height: Dimensions.get('window').height * 0.5,
    position: 'absolute',
    top: Dimensions.get('window').height * 0.25, 
    left: Dimensions.get('window').width * 0.1,
    borderRadius: 20,
    overflow: 'hidden',
    padding: 20,
    alignItems: 'center',
    backgroundColor: '#DCDCDC',
    justifyContent: 'space-around'
  },
  addTacoModalText: {
    fontSize: 24,
  },
  buttonStyle: {
    height: 50, 
    backgroundColor: '#00BFFF', 
    width: 200, 
    justifyContent: 'center', 
    alignItems: 'center', 
    borderRadius: 30
  }
})


