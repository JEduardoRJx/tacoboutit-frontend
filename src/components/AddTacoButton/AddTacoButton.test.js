import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as api from '../../apiCalls';

import AddTacoButton from '../AddTacoButton/AddTacoButton';

describe('AddTacoButton', () => {
  const mockSubmitTaco = jest.fn();
  it('renders correctly', () => {
    const wrapper = renderer.create(<AddTacoButton />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('handleSubmit resets state and calls submitTaco with correct params', async () => {
    const wrapper = shallow(<AddTacoButton submitTaco={mockSubmitTaco} id={1}/>);
    wrapper.setState({ showAddTaco: true, selectedType: 'test' })
    expect(wrapper.state('showAddTaco')).toEqual(true);
    await wrapper.instance().handleSubmit();
    expect(wrapper.state('showAddTaco')).toEqual(false);
    expect(mockSubmitTaco).toHaveBeenCalledWith('test', 1);
  });
});