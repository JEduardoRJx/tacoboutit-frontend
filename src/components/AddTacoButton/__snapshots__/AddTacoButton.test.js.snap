// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTacoButton enzyme snapshot test 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightgray",
      "borderRadius": 20,
      "height": 40,
      "justifyContent": "center",
      "position": "absolute",
      "right": 10,
      "top": 220,
      "width": 100,
      "zIndex": 5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    + Taco
  </Text>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#DCDCDC",
          "borderRadius": 20,
          "height": 667,
          "justifyContent": "space-around",
          "left": 75,
          "overflow": "hidden",
          "padding": 20,
          "position": "absolute",
          "top": 333.5,
          "width": 600,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Add New Taco
      </Text>
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Select Type
      </Text>
      <Picker
        mode="dialog"
        onValueChange={[Function]}
        selectedValue="carne asada"
        style={
          Object {
            "height": 50,
            "width": "100%",
          }
        }
      >
        <PickerItem
          label="Carne Asada"
          value="carne asada"
        />
        <PickerItem
          label="Al Pastor"
          value="al pastor"
        />
        <PickerItem
          label="Cabeza"
          value="cabeza"
        />
        <PickerItem
          label="Cabeza"
          value="cabeza"
        />
        <PickerItem
          label="Carnitas"
          value="carnitas"
        />
      </Picker>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00BFFF",
              "borderRadius": 30,
              "height": 50,
              "justifyContent": "center",
              "width": 200,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Submit
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </Component>
</TouchableOpacity>
`;

exports[`AddTacoButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightgray",
      "borderRadius": 20,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "right": 10,
      "top": 220,
      "width": 100,
      "zIndex": 5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    + Taco
  </Text>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#DCDCDC",
          "borderRadius": 20,
          "height": 667,
          "justifyContent": "space-around",
          "left": 75,
          "overflow": "hidden",
          "padding": 20,
          "position": "absolute",
          "top": 333.5,
          "width": 600,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Add New Taco
      </Text>
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Select Type
      </Text>
      <View
        style={
          Object {
            "height": 50,
            "width": "100%",
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "Carne Asada",
                "textColor": undefined,
                "value": "carne asada",
              },
              Object {
                "label": "Al Pastor",
                "textColor": undefined,
                "value": "al pastor",
              },
              Object {
                "label": "Cabeza",
                "textColor": undefined,
                "value": "cabeza",
              },
              Object {
                "label": "Cabeza",
                "textColor": undefined,
                "value": "cabeza",
              },
              Object {
                "label": "Carnitas",
                "textColor": undefined,
                "value": "carnitas",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00BFFF",
              "borderRadius": 30,
              "height": 50,
              "justifyContent": "center",
              "width": 200,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
